docs/source/building/variants.rst

```diff
@@ -0,0 +1,911 @@
+
+
+Special variant keys
+--------------------
+
+
+* ``pin_run_as_build``: should be a dictionary. Keys are package names. Values
+  are "pinning expressions" - explained in more detail in `Customizing
+  compatibility`_. This is a generalization of the ``numpy x.x`` spec.
+* ``extend_keys``: specifies keys that should be aggregated, rather than
+  clobbered, by later variants. These are detailed below in the `Extended keys`_
+  section.
+* ``runtimes``: detailed further in `Extra Jinja2 functions`_.
+* ``exclude_from_build_hash``: list of package names to exclude from
+  requirements/build when computing hash. Described further in `Avoiding
+  unnecessary builds`_.
+
+
+Avoiding unnecessary builds
+---------------------------
+
+
+To avoid building variants of packages where pinning does not necessitate having
+different builds, you can use the ``exclude_from_build_hash`` key in your
+variant. The way this works is that all variants are evaluated, but if any
+hashes are the same, then they are considered duplicates, and are deduplicated.
+By omitting some packages from the build dependencies, we can avoid creating
+unnecessarily specific hashes, and allow this deduplication.
+
+For example, let's consider a package that uses numpy in both run and build
+requirements, and a variant that includes two numpy versions:
+
+
+.. code-block:: python
+
+    variants = [{'numpy': ['1.10', '1.11']}]
+
+meta.yaml:
+
+.. code-block:: yaml
+
+   requirements:
+       build:
+           - numpy {{ numpy }}
+       run:
+           - numpy
+
+Here, the variant says that we'll have two builds - one for each numpy version.
+However, since this recipe does not pin numpy's run requirement (because it
+doesn't utilize numpy's C API), it is unnecessary to build it against both numpy
+1.10 and 1.11.  This example also assumes that numpy is not set in
+``pin_run_as_build``.
+
+Defaults for ``exclude_from_build_hash`` are ['numpy', 'mkl'].  These result in
+just one build.  The actual build performed is probably done with the last
+'numpy' list element in the variant, but that's more of an implementation detail
+that you should not depend on.  The order is considered unspecified behavior,
+because the output should be independent of the input versions.  If the output is
+not independent of input versions, don't use this key!
+
+Any pinning done in the run requirements will affect the hash, and thus builds will
+be done for each variant in the matrix.
+
+
+CONDA_* variables and command line arguments to conda-build
+-----------------------------------------------------------
+
+To ensure legacy consistency, environment variables such as CONDA_PY behave as
+they always have, and they clobber all variants set in files or passed to the
+API.
+
+The full list of respected environment variables are:
+
+* CONDA_PY
+* CONDA_NPY
+* CONDA_R
+* CONDA_PERL
+* CONDA_LUA
+
+Legacy CLI flags are also still available. These are sticking around for their
```

Comment:
I agree that legacy seems like the wrong word. It might be worth, somewhere in the larger set of conda build documentation, pointing out this simple way of pinning.